File: src/main/java/ru/fullrest/mfr/plugins_configuration_utility/model/repository/ReleaseRepository.java
package ru.fullrest.mfr.plugins_configuration_utility.model.repository;

import org.springframework.data.repository.CrudRepository;
import ru.fullrest.mfr.plugins_configuration_utility.model.entity.Group;
import ru.fullrest.mfr.plugins_configuration_utility.model.entity.Release;

import java.util.List;

/**
 * Created on 02.11.2018
 *
 * @author Alexey Plekhanov
 */
public interface ReleaseRepository extends CrudRepository<Release, Integer> {
    List<Release> findAllByGroup(Group group);

    List<Release> findAllByGroupAndAppliedIsTrue(Group group);
}

File: desktop/application/src/main/kotlin/ru/fullrest/mfr/plugins_configuration_utility/model/repository/ReleaseRepository.kt
package ru.fullrest.mfr.plugins_configuration_utility.model.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import ru.fullrest.mfr.plugins_configuration_utility.model.entity.Group
import ru.fullrest.mfr.plugins_configuration_utility.model.entity.Release

interface ReleaseRepository : JpaRepository<Release, Int> {
    fun findAllByGroup(group: Group): List<Release>

    fun findFirstByGroupAndAppliedIsTrue(group: Group): Release

    fun findAllByAppliedIsTrue(): List<Release>

    @Query("UPDATE Release r SET r.applied = :apply WHERE r = :release")
    fun updateIsApliedByRelise(release: Release, apply: Boolean)
}